src/tests/performance/get_trans.rs

```diff
@@ -0,0 +1,241 @@
+    // │      75 │        150 │          0 │        266 │              4 │          2 │          2 │          2 │          3 │         10 │         100.00 │      13.85 │       812.22 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     100 │        150 │          0 │        133 │              4 │          1 │          2 │          2 │          3 │         11 │          68.00 │      19.47 │       523.90 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     125 │        150 │          0 │        753 │              8 │          3 │          3 │          3 │          3 │         11 │          68.00 │      34.34 │       371.28 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     150 │        150 │          0 │       4146 │            372 │          3 │          3 │          3 │          4 │       2014 │          58.00 │      38.47 │       339.23 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     200 │        150 │          0 │      13640 │            487 │          2 │          3 │          3 │          3 │         94 │          44.50 │      48.56 │       274.92 │
+    // ├─────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼────────────┼────────────┼────────────┼────────────┼────────────────┼────────────┼──────────────┤
+    // │     250 │        150 │          0 │      22103 │            373 │          3 │          3 │          3 │          4 │         81 │          34.00 │      58.25 │       218.87 │
+    // └─────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴────────────┴────────────┴────────────┴────────────┴────────────────┴────────────┴──────────────┘
+    // *NOTE* run with `cargo test --release tests::performance::get_transaction -- --nocapture`
+    // Before running test generate dummy devices with different ips using toos/ips.py
+
+    let synth_counts = vec![1, 10, 20, 50, 75, 100, 125, 150, 200, 250];
+
+    let mut table = RequestsTable::default();
+
+    for synth_count in synth_counts {
+        let target = TempDir::new().expect("Unable to create TempDir");
+        let mut node = Node::builder()
+            .max_peers(MAX_PEERS)
+            .start(target.path(), NodeType::Stateful)
+            .await
+            .unwrap();
+        let node_addr = node.addr();
+
+        let mut synth_sockets = Vec::with_capacity(synth_count);
+        let mut ips = IPS.to_vec();

```

Comment:
nit: since we only `pop` from the list of `ips`, we might as well iterate as follows:
```
for (_, ip_addr) in (0..synth_count).zip(IPS)
```

Reply:
The IPS list can be empty or contain less entries than we'd like to spawn threads. In such case we use localhost addresses. When iterators are zipped the loop would not work if IPS is empty and not produce localhost addresses.